GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -625,7 +632,11 @@ private FileSystemTaskResult ExecuteBackgroundOperation(FileSystemTask gitUpdate
                     result = FileSystemTaskResult.Success;
                     if (!string.IsNullOrEmpty(gitUpdate.OldVirtualPath) && !IsPathInsideDotGit(gitUpdate.OldVirtualPath))
                     {
-                        result = this.AddModifiedPathAndRemoveFromPlaceholderList(gitUpdate.OldVirtualPath);
+                        result = this.TryRemoveModifiedPath(gitUpdate.OldVirtualPath, isFolder: false);
+                        if (result == FileSystemTaskResult.Invalid)

```

Comment:
Our general pattern is that enum values of `Invalid` are meant to be illegal, and you should throw if you ever see that value. Please create a new enum value if you intend to communicate some sort of error condition here. And actually from reading this code, I can't tell what that condition is, so a more descriptive name would be very helpful.

Reply:
Nevermind on creating a new value, see other comments below.